spring-pulsar/src/main/java/org/springframework/pulsar/core/reactive/ReactivePulsarSenderOperations.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.core.reactive;
+
+import org.apache.pulsar.client.api.MessageId;
+import org.apache.pulsar.client.api.MessageRouter;
+
+import reactor.core.publisher.Mono;
+
+/**
+ * The Pulsar reactive send operations contract.
+ *
+ * @param <T> the message payload type
+ * @author Christophe Bornet
+ */
+public interface ReactivePulsarSenderOperations<T> {
+
+	/**
+	 * Sends a message to the default topic in a reactive manner.
+	 * @param message the message to send
+	 * @return the id assigned by the broker to the published message
+	 */
+	Mono<MessageId> send(T message);
+
+	/**
+	 * Sends a message to the specified topic in a reactive manner.
+	 * @param topic the topic to send the message to or {@code null} to send to the
+	 * default topic
+	 * @param message the message to send
+	 * @return the id assigned by the broker to the published message
+	 */
+	Mono<MessageId> send(String topic, T message);
+
+	/**
+	 * Create a {@link ReactiveSendMessageBuilder builder} for configuring and sending a
+	 * message reactively.
+	 * @param message the payload of the message
+	 * @return the builder to configure and send the message
+	 */
+	ReactiveSendMessageBuilder<T> newMessage(T message);
+
+	/**
+	 * Builder that can be used to configure and send a message. Provides more options
+	 * than the send methods provided by {@link ReactivePulsarSenderOperations}.
+	 *
+	 * @param <T> the message payload type
+	 */
+	interface ReactiveSendMessageBuilder<T> {
```

Comment:
I was going to save this comment as overall for the PR but I think it fits well here. 

The Reactive client has the concept of ReactiveMessageSender / ReactiveMessageSenderBuilder and MessageSpec / MessageSpecBuilder. When combined w/ the spring-pulsar concepts of ReactiveSendMessageBuilder and ReactivePulsarSenderFactory keeping track of whats-what was a bit confusing (for me) **at first**.  However, after digesting a bit, I think this is just the nature of things and the names you have chosen all make sense and line up w/ what was done for the non-reactive counterparts. 

That being said, I wonder if removing the `Reactive` prefix from `ReactiveSendMessageBuilder` -> `SendMessageBuilder` would simplify things a bit. I think we are safe from name collisions via packages. Wdyt?

Reply:
Sure, I'll rename it.

Reply:
done